### PR TITLE
Add depreciation warnings

### PR DESCRIPTION
**Pull request by sean-m-sullivan**
# What does this PR do?
Address #293 and add depreciation warnings for 4 modules
This will show up when the module is run, either directly or through the roles, warnings mention both.

# How should this be tested?
The floor is lava

# Is there a relevant Issue open for this?

# Other Relevant info, PRs, etc
#293

### CODE REVIEW

File: changelogs/fragments/depreciation.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+minor_changes:
```
Comment (Tompage1994) [inline changelogs/fragments/depreciation.yml]: There is a heading in the changelog for this:

```suggestion
deprecated_features:
```

File: plugins/modules/ah_ee_namespace.py
Comment (Tompage1994) [inline plugins/modules/ah_ee_namespace.py:1]: You also need to add to the documentation of the module that it is being deprecated. [This page](https://docs.ansible.com/ansible/latest/dev_guide/module_lifecycle.html#module-lifecycle) explains it but you need to add:

```
deprecated:
  removed_in: <version target>
  removed_by_date: <date>  # Mutually exclusive with removed_in 
  why: <optional>
  alternative: <optional>  # Something like: Use M(x.y.z) instead.
```

My suggestion is we go for something here like:

```
deprecated:
  removed_by_date: November 2023
  why: The endpoint has been removed and is not supported in AAP 2.3 onwards
  alternative: Use M(<alt module>) instead.
```

Alternatively we can go for `removed_in: 3.0.0` as we will be releasing 2.0.0 soon and we will need to major bump the version when removing a module.

File: plugins/modules/ah_group_perm.py [hunk 2]
```diff
@@ -165,6 +165,9 @@ def main():
 
     # Create a module for ourselves
     module = AHAPIModule(argument_spec=argument_spec, supports_check_mode=True)
+    module.warn("This module 'ah_group_perm' and the group permision part of the role 'group' will be removed when support for AAP 2.2 "
+                "ends in November of 2023. It was repplaced with role permisions in 2.3."
```
Comment (Tompage1994) [inline plugins/modules/ah_group_perm.py]:
```suggestion
                "ends in November of 2023. It was replaced with role permissions in 2.3."
```

File: plugins/modules/ah_ee_namespace.py [hunk 3]
```diff
@@ -208,6 +208,9 @@ def main():
 
     # Create a module for ourselves
     module = AHAPIModule(argument_spec=argument_spec, supports_check_mode=True)
+    module.warn("This role 'ee_namespace' and module 'ah_ee_namespace' will be removed when support for AAP 2.2 ends in November of 2023. "
+                "It was removed in 2.3."
```
Comment (Tompage1994) [inline plugins/modules/ah_ee_namespace.py]: Haven't checked but is there some endpoint to replace this? Is this a new module we need to write?

File: plugins/modules/ah_ee_namespace.py
Comment (Tompage1994) [reply plugins/modules/ah_ee_namespace.py:1]: Obviously needed for each module

**Review by Tompage1994 — CHANGES_REQUESTED** [review]
A few changes and suggestions

File: plugins/modules/ah_ee_namespace.py (hunk 3 above)
Comment (sean-m-sullivan) [reply plugins/modules/ah_ee_namespace.py]: No, it was just completely removed, outright, no replacement

File: plugins/modules/ah_ee_namespace.py [hunk 4]
```diff
@@ -24,6 +29,10 @@
   - Please use the ee_repository to achieve the same functionality.
 version_added: '0.4.3'
 author: Herve Quatremain (@herve4m)
+deprecated:
+  removed_by_date: November 2023
+  why: The endpoint has been removed and is not supported in AAP 2.3 onwards
+  alternative: Use M(<alt module>) instead.
```
Comment (Tompage1994) [inline plugins/modules/ah_ee_namespace.py]: Is it better to just say there is no alternative at present?

**Review by Tompage1994 — APPROVED** [review]
Just one comment and up to you whether to change it

File: plugins/modules/ah_ee_namespace.py (hunk 4 above)
Comment (sean-m-sullivan) [reply plugins/modules/ah_ee_namespace.py]: I'm working on getting these working

**Comment by sean-m-sullivan** [comment]
Okay I think I have it now. This passes sanity,

**Review by djdanielsson — APPROVED** [review]
LGTM

**Pull request merged**